test-integration/utils/actions/escrow.js

```diff
@@ -46,58 +77,108 @@ const claimTokensByLoanId = async (
+};
+
+const callDapp = async (
+  { escrow },
+  { txConfig, testContext },
+  { dappName, fnName, args }
 ) => {
-  const claimTokensResult = await escrow.claimTokens(recipient, txConfig);
-  escrowEvents
-    .tokensClaimed(claimTokensResult)
-    .emitted(recipient);
-  return claimTokensResult;
+  const { getContracts } = testContext;
+  const { address: location, contract: dappContract } = await getContracts.getDeployed(
+    teller.escrowDapp(dappName)
+  );
+  const dappData = {
+    location,
+    data: dappContract.methods[fnName](...args).encodeABI()
+  };
+  return escrow.callDapp(dappData, txConfig);
 };
 
 /**
  * Use the Uniswap Dapp to swap a token using the funds in an Escrow.
+ * @returns number - Balance of the destination token if swap was successful
  */
 const uniswapSwap = async (
   { escrow },
   { txConfig, testContext },
-  { path, sourceAmount, minDestination }
+  { tokensPath, sourceAmount, minDestination, shouldFail = false, expectedRevertReason }
 ) => {
-  const { getContracts } = testContext;
-  const { address: uniswapDappAddress, contract: uniswapDapp  } = await getContracts.getDeployed(
-    teller.escrowDapp(logicNames.Uniswap)
+  const tokenA = tokensPath[0];
+  const tokenB = tokensPath[tokensPath.length - 1];
+  const tokenABalanceBefore = (await tokenA.balanceOf(escrow.address)).toString();
+  const tokenBBalanceBefore = (await tokenB.balanceOf(escrow.address)).toString();
+
+  const path = tokensPath.map(({ address }) => address);
+  const args = [ path, sourceAmount, minDestination ];
+  const fn = callDapp(
+    { escrow },
+    { txConfig, testContext },
+    { dappName: logicNames.Uniswap, fnName: "swap", args }
   );
-  const dappData = {
-    location: uniswapDappAddress,
-    data: uniswapDapp.methods.swap(path, sourceAmount, minDestination).encodeABI()
-  };
-  const swapResult = await escrow.callDapp(dappData, txConfig);
 
-  // TODO Add assertions
-  // const sourceTokenAddress = path[0];
-  // const destinationTokenAddress = path[path.length - 1];
-  // uniswapEvents
-  //   .uniswapSwapped(swapResult)
-  //   .emitted(txConfig.from, escrow.address, sourceTokenAddress, destinationTokenAddress, sourceAmount);
+  if (shouldFail) {
+    try {
+      await assert.rejects(fn, "Expected swap to fail");
+    } catch (error) {
+      assert.strictEqual(error.reason, expectedRevertReason, error.message);
+    }
+  } else {
+    await assert.doesNotReject(fn, "Expected swap to be successful");
 
-  return swapResult;
+    await tokensAssertions.balanceLt(
+      { token: tokenA },
+      { testContext },
+      { address: escrow.address, maxBalance: tokenABalanceBefore }
+    );
+    const minDestinationBalance = new BigNumber(tokenBBalanceBefore).plus(minDestination);
+    await tokensAssertions.balanceGt(
+      { token: tokenB },
+      { testContext },
+      { address: escrow.address, minBalance: minDestinationBalance.toString() }
+    );
+
+    // const sourceTokenAddress = path[0];
```

Comment:
why is this code commented? Do we need to remove it?

Reply:
It needs to be updated to get the transaction receipt and then get the logs to see if the event was emitted.

Reply:
Okay, please add a TODO